zebracrossing/campaigns/views.py

```diff
@@ -1,26 +1,55 @@
+from curses.ascii import HT
 from datetime import date, timedelta
 import mimetypes
-
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.contrib.auth.decorators import login_required
 from django.contrib.auth import mixins
 from django.shortcuts import get_object_or_404, render
-from django.http import HttpResponse
+from django.http import HttpResponse, HttpResponseBadRequest
 from django.views.generic import DetailView
 from django.views.generic.edit import CreateView
 from django.urls import reverse
 
-from .models import BookingSheet, Campaign, Material
+from .models import BookingSheet, Campaign, Material, BookedDay
 from .forms import BookingSheetForm, CampaignForm, MaterialForm
+from rest_framework.views import APIView
+from rest_framework.response import Response
 
 
 class CampaignView(mixins.LoginRequiredMixin, DetailView):
     model = Campaign
 
+    def schedule_data(self, **kwargs):
+        sheet = BookingSheet.objects.filter(campaign=self.get_object())[0]
+        return sheet
+
 
 class BookingView(mixins.LoginRequiredMixin, DetailView):
     model = BookingSheet
 
 
+class scheduleList(LoginRequiredMixin, APIView):
+    def get(self, request):
+        start_date = request.GET.get("start")
+        end_date = request.GET.get("end")
+        schedule_list = []
+        if start_date is None or end_date is None:
+            return HttpResponseBadRequest(
+                "Please check that your start and end date are valid and present",
+                status=400,
+            )
+        else:
```

Comment:
Doesn't need to be within `else`, as the return above will prevent that code path from getting here